### DISCUSSION TITLE
NS Panel not started

### DISCUSSION THREAD

**Discussion by gonzzzales** [arendst/Tasmota#22519] [Q&A]
Hi, i have some problems with my NSPanel. The panel does not want to run autoexec.be
Here is my web log (Debug level)

00:00:00.003 HDW: ESP32-D0WD-V3 (PSRAM)
00:00:00.004 HDW: FoundPSRAM=1 CanUsePSRAM=1
00:00:00.182 UFS: FlashFS mounted with 228 kB free
00:00:00.291 CFG: Loaded from File, Count 681
00:00:00.307 QPC: Count 1
00:00:00.309 CFG: CR 458/699, Busy 0
00:00:00.318 CFG: CR 458/699, Busy 0
00:00:00.321 ROT: Mode 1
00:00:00.335 BRY: GC from 3699 to 3067 bytes, objects freed 5/22 (in 1 ms) - slots from 34/61 to 23/61
00:00:00.685 CFG: No '*.autoconf' file found
00:00:00.690 BRY: GC from 4477 to 3731 bytes, objects freed 5/38 (in 1 ms) - slots from 55/76 to 35/76
00:00:00.691 BRY: Berry initialized, RAM used=3731 bytes
00:00:00.717 BRY: No 'preinit.be'
00:00:00.721 SRC: Restart
00:00:00.728 Project tasmota - Tasmota Version 11.1.0.2(tasmota)-2_0_3(2022-05-13T09:12:35)
00:00:00.729 ETH: No ETH MDC and/or ETH MDIO GPIO defined
00:00:00.746 DSB: Sensors found 0
00:00:01.277 BRY: GC from 8039 to 6862 bytes, objects freed 19/60 (in 2 ms) - slots from 87/122 to 62/122
00:00:01.328 BRY: GC from 13764 to 13764 bytes, objects freed 0/234 (in 1 ms) - slots from 211/244 to 211/244
00:00:01.436 BRY: GC from 28371 to 28371 bytes, objects freed 0/449 (in 2 ms) - slots from 380/411 to 380/411
00:00:01.701 SRC: Berry
00:00:01.703 CMD: Grp 0, Cmd 'TOPIC', Idx 1, Len 0, Pld -99, Data ''
00:00:01.706 RSL: RESULT = {"Topic":"NSP_Kuhnya"}
00:00:01.712 NXP: Initializing Driver
00:00:01.719 NXP: Screen Initialized
00:00:02.335 BRY: GC from 56777 to 42871 bytes, objects freed 149/701 (in 25 ms) - slots from 623/654 to 559/593
00:00:02.368 BRY: failed to run compiled code 'key_error' - 44
00:00:02.369 BRY: No 'autoexec.be'
00:00:03.376 WIF: Checking connection...
00:00:03.377 WIF: Attempting connection...
00:00:03.877 WIF: Connecting to AP1 Cluster Channel 1 BSSId 50:FF:20:70:43:B2 in mode 11n as NSP-Kuhnya-0748...
00:00:06.460 WIF: Checking connection...
00:00:06.462 WIF: Connected

I am concerned about the following entries
00:00:00.685 CFG: No '*.autoconf' file found
00:00:00.717 BRY: No 'preinit.be'
00:00:02.368 BRY: failed to run compiled code 'key_error' - 44
00:00:02.369 BRY: No 'autoexec.be'
Of course the file autoexec.be exists.
A complete reinstallation of Tasmota helps, but the situation has been repeated several times already and i want to understand the reasons.
Please tell me what can be done about it??

**Comment by sfromis** [comment]
It **was** running `autoexec.be`, but failing with
`BRY: failed to run compiled code 'key_error' - 44`
"Someone" would have to debug details of that situation. Plausibly, the code was written with an assumption which was no longer met.

In case of `weatherCode` from `wttr.in` having an unexpected value of 44, that message would be possible, but I'm not saying that this is what happened, and it does not look like 44 should be among the 3-digit weather codes from that service.

**Comment by gonzzzales** [comment]
"Someone" is ready to debug it)))
Thanks for the tip, I thought the problem was in Tasmota (tried googling "error code - 44"). When I placed the debugging messages in autoexec.be, I immediately found the problem. May be it is better to change the message in the log, something like "In autoexec.be runtime error, please check it"
